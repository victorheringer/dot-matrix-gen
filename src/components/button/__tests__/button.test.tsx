import { render } from "@testing-library/react";
import { Button } from "../index";

describe("Button", () => {
  it("should render correctly the default variant", () => {
    const tree = render(<Button variant="default" />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly the danger variant", () => {
    const tree = render(<Button variant="danger" />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly the secondary variant", () => {
    const tree = render(<Button variant="secondary" />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly the default variant and full width", () => {
    const tree = render(<Button variant="default" full />).asFragment();
    expect(tree).toMatchSnapshot();
  });
});
