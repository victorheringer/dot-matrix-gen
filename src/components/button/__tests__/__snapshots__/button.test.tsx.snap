// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly the danger variant 1`] = `
<DocumentFragment>
  <button
    class="sc-bczRLJ bzCXpw"
  />
</DocumentFragment>
`;

exports[`Button should render correctly the default variant 1`] = `
<DocumentFragment>
  <button
    class="sc-bczRLJ dTuaoz"
  />
</DocumentFragment>
`;

exports[`Button should render correctly the default variant and full width 1`] = `
<DocumentFragment>
  <button
    class="sc-bczRLJ bZsGtM"
  />
</DocumentFragment>
`;

exports[`Button should render correctly the secondary variant 1`] = `
<DocumentFragment>
  <button
    class="sc-bczRLJ diUefb"
  />
</DocumentFragment>
`;
